kernel/aster-nix/src/util/net/options/utils.rs

```diff
@@ -141,6 +141,42 @@ impl WriteToUser for LingerOption {
     }
 }
 
+impl ReadFromUser for Duration {
+    fn read_from_user(vmar: &Vmar<Full>, addr: Vaddr, max_len: u32) -> Result<Self> {
+        if (max_len as usize) < core::mem::size_of::<Duration>() {
+            return_errno_with_message!(Errno::EINVAL, "max_len is too short");
+        }
+        let mut bytes = vec![0; max_len as usize];
+        vmar.read_bytes(addr, &mut bytes)?;
+
+        let secs_bytes: [u8; 8] = bytes[0..8].try_into().unwrap();
+        let nanos_bytes: [u8; 4] = bytes[8..12].try_into().unwrap();
+
+        let secs = u64::from_ne_bytes(secs_bytes);
+        let nanos = u32::from_ne_bytes(nanos_bytes);
+
+        Ok(Duration::new(secs, nanos))
+    }
```

Comment:
`Duration` has implemented `From<timeval_t>` trait. So the `read_from_user` can be refactored as
```rust
fn read_from_user(vmar: &Vmar<Full>, addr: Vaddr, max_len: u32) -> Result<Self> {
    if (max_len as usize) < core::mem::size_of::<timeval_t>() {
        return_errno_with_message!(Errno::EINVAL, "max_len is too short");
    }

    let timeval: timeval_t = vmar.read_val(addr)?;
    Ok(Duration::from(timeval))
}
```
The `write_to_user` should be refactored similarly.

Reply:
> Avoids using C type like `timeval_t` in `net` module. C types are supposed to only be used in `syscall` module and should be converted to Rust type when calling functions in submodules.
> 
> So these two structs should be defined as
> 
> ```rust
> pub struct RcvTimeoOld(Duration);
> pub struct SndTimeoOld(Duration);
> ```

But I assume using `timeval_t` in net is being frown upon? I guess it would be easier to cast from the `timeval_t` so that any update related to Duration can be automatically sync here.

Reply:
> But I assume using timeval_t in net is being frown upon? I guess it would be easier to cast from the timeval_t so that any update related to Duration can be automatically sync here.

Sorry, I didn't understand your words. Are you suggesting that using `timeval_t` in the `net` module would be more convenient?